pkg/tls/tls.go

```diff
@@ -52,3 +57,59 @@ func MakeCA(signerName string) (*crypto.CA, error) {
 		Config:          caConfig,
 	}, nil
 }
+
+const (
+	TigeraCASecretName          = "tigera-ca-private"
+	TrustedCertConfigMapName    = "tigera-ca-bundle"
+	TrustedCertConfigMapKeyName = "tigera-ca-bundle.crt"
+	TrustedCertVolumeMountPath  = "/etc/pki/tls/certs/"
+	TrustedCertBundleMountPath  = "/etc/pki/tls/certs/tigera-ca-bundle.crt"
+)
+
+// TigeraCA is a signer of new certificates and has methods to retrieve existing KeyPairs and Certificates. If a user
+// brings their own secrets, TigeraCA will preserve and return them.
+type TigeraCA interface {
+	// CertificateManagement returns the CertificateManagement object or nil if it is not configured.
+	CertificateManagement() *operatorv1.CertificateManagement
+	// GetKeyPair returns an existing KeyPair. If the KeyPair is not found, a k8s.io NotFound error is returned.
+	GetKeyPair(cli client.Client, secretName, secretNamespace string) (KeyPair, error)
```

Comment:
Is `NotFound` always going to be treated as an error but the callers? Can we swallow the `NotFound` and just return nil?